sqldelight-compiler/src/main/kotlin/com/squareup/sqldelight/SqlitePluginException.kt

```diff
@@ -18,4 +18,8 @@ package com.squareup.sqldelight
 import org.antlr.v4.runtime.ParserRuleContext
 
 class SqlitePluginException(val originatingElement: ParserRuleContext, override val message: String)
-    : IllegalStateException(message)
+    : IllegalStateException(message) {
+  override fun hashCode() = originatingElement.sourceInterval.hashCode()
+  override fun equals(other: Any?) = other is SqlitePluginException &&
+      originatingElement.sourceInterval.equals(other.originatingElement.sourceInterval)
```

Comment:
Not a huge fan of this since you're technically violating the contract of what this is supposed to do. Can you use a multimap from element to list/set of exceptions instead?